File: 2455-average-value-of-even-numbers-that-are-divisible-by-three/2455-average-value-of-even-numbers-that-are-divisible-by-three.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    int averageValue(vector<int>& nums) {
        vector<int> s_nums;
        copy_if(nums.begin(), nums.end(), back_inserter(s_nums), [](int num) {
            return num % 6 == 0;
        });
        int sumx = accumulate(s_nums.begin(), s_nums.end(), 0);
        return s_nums.empty() ? 0 : sumx / s_nums.size();
    }
};

File: 2455-average-value-of-even-numbers-that-are-divisible-by-three/2455-average-value-of-even-numbers-that-are-divisible-by-three.rb
# @param {Integer[]} nums
# @return {Integer}
def average_value(nums) # num.even? && num % 3 == 0 simplified below
    selected_nums = nums.select{|num| num % 6 == 0}
    selected_nums.empty? ? 0 : selected_nums.sum / selected_nums.size
end